tests/study_tests/test_study.py

```diff
@@ -1122,6 +1122,68 @@ def test_ask_fixed_search_space() -> None:
     assert params["y"] in ["bacon", "spam"]
 
 
+# Deprecated distributions are internally converted to corresponding distributions.
+def test_ask_distribution_conversion() -> None:
+    fixed_distributions = {
+        "ud": distributions.UniformDistribution(low=0, high=10),
+        "dud": distributions.DiscreteUniformDistribution(low=0, high=10, q=2),
+        "lud": distributions.LogUniformDistribution(low=1, high=10),
+        "id": distributions.IntUniformDistribution(low=0, high=10),
+        "idd": distributions.IntUniformDistribution(low=0, high=10, step=2),
+        "ild": distributions.IntLogUniformDistribution(low=1, high=10),
+    }
+
+    study = create_study()
+
+    with patch(
+        "optuna.distributions._warn_old_distribution_conversion",
+    ) as mock_obj:
+        trial = study.ask(fixed_distributions=fixed_distributions)
+        assert mock_obj.call_count == 6
+        assert mock_obj.call_args_list == [
+            (("UniformDistribution", "FloatDistribution"),),
+            (("DiscreteUniformDistribution", "FloatDistribution"),),
+            (("LogUniformDistribution", "FloatDistribution"),),
+            (("IntUniformDistribution", "IntDistribution"),),
+            (("IntLogUniformDistribution", "IntDistribution"),),
+        ]
```

Comment:
Sorry, I wasn't clear when I said to test if `ask` and `create_trial` are raising a warning. What i had in mind was something like:
```python
with pytest.warns(SomeWarning):
    trial = study.ask(fixed_distributions)
```
I don't think we need to patch stuff here.

Reply:
I'd say +1 for parametrization and avoid mocking where possible. Alternatively, instead of checking call count of mocked function, we could just test how many FutureWarnings were raised:

```python
with pytest.warns(RuntimeWarning) as record:
    warnings.warn("another warning", RuntimeWarning)

# check that only one warning was raised
assert len(record) == 1
# check that the message matches
assert record[0].message.args[0] == "another warning"
```

Reply:
Thank you for sharing the thought! I prefer the latter scenario. 838dd36 replace mocking stuff with pytest's record object.